File: Dev/C++/Mascaret/src/VEHA/Behavior/BasicActions/AcceptEventAction.cpp
#include "VEHA/Behavior/BasicActions/AcceptEventAction.h"
#include "VEHA/Behavior/BasicActions/AcceptEventBehaviorExecution.h"

namespace VEHA
{

AcceptEventAction::AcceptEventAction()
{
	_kind = "AcceptEvent";
}

AcceptEventAction::~AcceptEventAction()
{
}

shared_ptr<BehaviorExecution> AcceptEventAction::createBehaviorExecution(shared_ptr<InstanceSpecification> host, const Parameters& p, bool sync)
{
	shared_ptr<AcceptEventBehaviorExecution> behavior(new AcceptEventBehaviorExecution(shared_dynamic_cast<AcceptEventAction>(shared_from_this()),host,p));
	return behavior;
}

}

File: Dev/CS/Mascaret/Mascaret/VEHA/Behavior/BasicActions/AcceptEventAction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Mascaret
{
    class AcceptEventAction : Action, IEquatable<AcceptEventAction>
    {
        public AcceptEventAction()
        {
            Kind = "AcceptEvent";
        }


        public Trigger getTrigger()
        {
            return _trigger;
        }

        public void setTrigger(Trigger trigger)
        {
            _trigger = trigger;
        }

        public bool Equals(AcceptEventAction other)
        {
            if (other == this)
                return true;
            return false;
        }

        public override BehaviorExecution createBehaviorExecution(InstanceSpecification host, Dictionary<string, ValueSpecification> p, bool sync)
        {
            AcceptEventBehaviorExecution behavior = new AcceptEventBehaviorExecution(this, host, p);
            return behavior;
        }

        protected Trigger _trigger;
    }
}
